contrib/ci/build_windows.sh

```diff
@@ -57,25 +69,91 @@ echo $VERSION > $DESTDIR/VERSION
 sed -i 's,UpdateMotd=.*,UpdateMotd=false,' $DESTDIR/etc/fwupd/daemon.conf
 
 # create a setup binary
-cd $DESTDIR
-mkdir -p $DESTDIR/setup
-makensis -NOCD $build/contrib/setup-win32.nsi
+CERTDIR=/etc/pki/tls/certs
+MINGW32BINDIR=/usr/x86_64-w64-mingw32/sys-root/mingw/bin
 
-#so that it's actually executable
-cp /usr/x86_64-w64-mingw32/sys-root/mingw/bin/*.dll bin
+# deps
+find $MINGW32BINDIR \
+	-name curl.exe \
+	-o -name gspawn-win64-helper-console.exe \
+	-o -name gspawn-win64-helper.exe \
+	-o -name iconv.dll \
+	-o -name libarchive-13.dll \
+	-o -name libbrotlicommon.dll \
+	-o -name libbrotlidec.dll \
+	-o -name libbz2-1.dll \
+	-o -name libcrypto-1_1-x64.dll \
+	-o -name libcurl-4.dll \
+	-o -name libffi-*.dll \
+	-o -name libgcc_s_seh-1.dll \
+	-o -name libgio-2.0-0.dll \
+	-o -name libglib-2.0-0.dll \
+	-o -name libgmodule-2.0-0.dll \
+	-o -name libgmp-10.dll \
+	-o -name libgnutls-30.dll \
+	-o -name libgnutls-30.dll \
+	-o -name libgobject-2.0-0.dll \
+	-o -name libhogweed-*.dll \
+	-o -name libidn2-0.dll \
+	-o -name libintl-8.dll \
+	-o -name libjson-glib-1.0-0.dll \
+	-o -name liblzma-5.dll \
+	-o -name libnettle-*.dll \
+	-o -name libp11-kit-0.dll \
+	-o -name libpcre-1.dll \
+	-o -name libsqlite3-0.dll \
+	-o -name libssh2-1.dll \
+	-o -name libssl-1_1-x64.dll \
+	-o -name libssp-0.dll \
+	-o -name libtasn1-6.dll \
+	-o -name libusb-1.0.dll \
+	-o -name libwinpthread-1.dll \
+	-o -name libxml2-2.dll \
+	-o -name zlib1.dll \
+	| wixl-heat \
+	-p $MINGW32BINDIR/ \
+	--directory-ref BINDIR \
+	--var "var.MINGW32BINDIR" \
+	--component-group "CG.fwupd-deps" | \
+	tee $build/contrib/fwupd-deps.wxs
```

Comment:
although I really like how programattic this list is, this change breaks running without installing.  As it stands today you can drop a `dist` directory onto a real Windows host (or "natively" launch it from WSL2) and it runs.  I want to maintain that if at all possible.

Can we maybe make symlinks for all the files and just not install the symlinks from the MSI?

Reply:
would copying those files into `$DESTDIR/bin` also work? That does feel a bit of a hack. Can you describe a bit how you use WSL right now so I can make sure this doesn't break things?

Reply:
What i've been doing is:
1. Open up Fedora in WSL
2. `./contrib/ci/build_windows.sh`
3. `./dist/bin/fwupdtool.exe`

I don't mind changing that workflow a bit (add environment variable here, or there etc), but adding a step of installing an MSI is too much to me.

Reply:
Copying those files to `$DESTDIR/bin` would also work.  Or I wonder if maybe there is an easy way to just extract the MSI and run it from that directory it's extracted to?

Reply:
Or is it possible on Windows to set the DLL search path add /usr/x86_64-w64-mingw32/sys-root/mingw/bin`?

Reply:
Not easily that I can tell.   How about if you just run `msiextract` at the end of the script?  It should overwrite everything in dist and copy in the missing dependency libraries.

Reply:
That didn't work super-well. I've added a more useful script in ./contrib -- can you try that with WSL and Fedora please.

Reply:
I actually got WSL working with Fedora too. :)

Reply:
It's handy right!